### DISCUSSION TITLE
Additions to `.gitignore`

### DISCUSSION THREAD

**Discussion by elsandosgrande** [KDAB/hotspot#405] [General]
It would be nice to add some common files to `.gitignore`, like VS Code settings and workspace settings files, or clangd compilation database files. Specifically:
```gitignore
.vscode # directory used by VS Code for preferences and the like
compile_flags.txt # list of used compilation flags, handwritten
compile_commands.json # compilation database, generated by CMake; https://clangd.llvm.org/design/compile-commands#where-do-compile-commands-come-from
*.code-workspace # VS Code workspace file
.cache # clangd's index cache; https://clangd.llvm.org/design/indexing.html
```

Thank you in advance!!

**Comment by beedell-roke** [comment]
> https://github.com/KDAB/hotspot/discussions/405#discussion-4456116

@elsandosgrande, https://stackoverflow.com/a/5724484/9731176 might be better (or at least useful in the meantime).

**Comment by milianw** [comment]
patches for such things by regular contributors is welcome, but I'm not going to list all random IDEs that some people might be using "just because". Using a global personal gitignore on your end instead as suggested by @RokeJulianLockhart is a better approach for that.
